components/projects/projectmodel.js

```diff
@@ -0,0 +1,20 @@
+const mongoose = require('mongoose');
+
+// Define the schema
+const { Schema } = mongoose;
+
+const ProjectSchema = new Schema({
+  title: String,
+  authorId: Schema.Types.ObjectId,
+  authorName: String,
+  keywords: [String],
+  description: String,
+  lastSaved: Date,
```

Comment:
Thanks I'll look into the timestamps.

As a user, what are you expecting from a date of creation for the project? I am not sure what value it adds.

I think it would actually be a date of first publication, because if the owner sits on the draft for a month or two before publishing, that's none of the other users' business.

So when you look at a project, would you see the date of last (published) change and the date of first publication? What are the use cases for needing this? I'm wondering if it's a case for separating the project from the help wanted ad or if you mean something different.

Reply:
I guess date of first published would do something similar, but possibly a project owner might want to look back and see how long their project has been running on the site. Maybe just for a sense of accomplishment more than anything if it's a long running thing. Also users may want to see when a project was made, if it was created a long time ago with not many updated they might not want to join.

But first published covers both of those.